che/service/che_service.go

```diff
@@ -0,0 +1,81 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	token2 "github.com/fabric8-services/fabric8-auth/authorization/token"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-auth/application/service"
+	"github.com/fabric8-services/fabric8-auth/application/service/base"
+	servicecontext "github.com/fabric8-services/fabric8-auth/application/service/context"
+	"github.com/fabric8-services/fabric8-auth/authentication/account/repository"
+	"github.com/fabric8-services/fabric8-auth/authorization/token/manager"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	errs "github.com/pkg/errors"
+)
+
+// cheServiceImpl is the default implementation of CheService.
+type cheServiceImpl struct {
+	base.BaseService
+	config Configuration
+}
+
+// Configuration the config for the Che service
+type Configuration interface {
+	manager.TokenManagerConfiguration
+	GetCheServiceURL() string
+}
+
+// NewCheService creates a new Che service.
+func NewCheService(context servicecontext.ServiceContext, config Configuration) service.CheService {
+	return &cheServiceImpl{
+		BaseService: base.NewBaseService(context),
+		config:      config,
+	}
+}
+
+// DeleteUser deletes a user in Che
+func (s *cheServiceImpl) DeleteUser(ctx context.Context, identity repository.Identity) error {
+	log.Info(ctx, map[string]interface{}{"identity_id": identity.ID.String()}, "deleting user on Che service")
+	// this endpoint is restricted to the `auth` Service Account
+	deleteUserAPIURL := fmt.Sprintf("%s/api/user/%s", s.config.GetCheServiceURL(), identity.ID.String())
+	req, err := http.NewRequest("DELETE", deleteUserAPIURL, nil)
+	if err != nil {
+		return errs.Wrapf(err, "unable to delete user '%s' in Che", identity.ID.String())
+	}
+
+	tokenManager, err := manager.DefaultManager(s.config)
+	if err != nil {
+		return errs.Wrapf(err, "unable to delete user '%s' in Che", identity.ID.String())
+	}
+
+	token, err := tokenManager.GenerateTransientUserAccessTokenForIdentity(ctx, identity)
+	if err != nil {
+		return errs.Wrapf(err, "unable to delete user '%s' in Che", identity.ID.String())
+	}
+
+	_, err = s.Services().TokenService().RegisterToken(ctx, identity.ID, *token, token2.TOKEN_TYPE_ACCESS, nil)
```

Comment:
I thought the point of having a transient access token was because we did not want to register it? Or is it mandatory for the token validation? If so, could we have an extra claim field to flag the token as transient and skip the database validation?

Reply:
Yes, we have to handle it but I don't see it in this PR.
We either should skip the registration and validation completely or keep registration/validation but skip touching user activity field during validation.

Reply:
We already have a specified claim field for transient tokens ("transient"=true) however Alexey has made a good point about validation, this is something I overlooked so I'll address it ASAP.

Reply:
I've added a check to the `ValidateToken()` to skip updating the identity's last active timestamp if the token is transient.  There's still a failure in the user_blackbox_test though, for which I'm not sure how to fix yet.

Reply:
Strange.. the tests pass in CI even though they fail for me locally.  Not sure what to make of that, @xcoulon can you try running the tests locally?

Reply:
Ignore my previous comment, after cleaning up the database the tests now pass successfully.  It seems that Alexey is still experiencing a test failure though, so my point still stands @xcoulon that it would be interesting to see what result you get.

Reply:
@sbryzak see https://github.com/fabric8-services/fabric8-auth/pull/816 as a separate PR to address this problem (as I could reproduce it locally on `master` branch as well)